### DISCUSSION TITLE
JSON shield giving invalid result

### DISCUSSION THREAD

**Discussion by DavidStirling** [badges/shields#6852] [Q&A] [answered]
One of our badges recently stopped working - [![Image.sc forum](https://img.shields.io/badge/dynamic/json.svg?label=forum&url=https%3A%2F%2Fforum.image.sc%2Ftag%2Fcellprofiler.json&query=%24.topic_list.tags.0.topic_count&colorB=brightgreen&suffix=%20topics&logo=data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABPklEQVR42m3SyyqFURTA8Y2BER0TDyExZ+aSPIKUlPIITFzKeQWXwhBlQrmFgUzMMFLKZeguBu5y+//17dP3nc5vuPdee6299gohUYYaDGOyyACq4JmQVoFujOMR77hNfOAGM+hBOQqB9TjHD36xhAa04RCuuXeKOvwHVWIKL9jCK2bRiV284QgL8MwEjAneeo9VNOEaBhzALGtoRy02cIcWhE34jj5YxgW+E5Z4iTPkMYpPLCNY3hdOYEfNbKYdmNngZ1jyEzw7h7AIb3fRTQ95OAZ6yQpGYHMMtOTgouktYwxuXsHgWLLl+4x++Kx1FJrjLTagA77bTPvYgw1rRqY56e+w7GNYsqX6JfPwi7aR+Y5SA+BXtKIRfkfJAYgj14tpOF6+I46c4/cAM3UhM3JxyKsxiOIhH0IO6SH/A1Kb1WBeUjbkAAAAAElFTkSuQmCC)](https://forum.image.sc/tag/cellprofiler)

It's supposed to fetch a topic count from a forum running on Discourse. The JSON URL seems to work (`https://forum.image.sc/tag/cellprofiler.json`) and to my understanding the expression should be `$topic_list.tags.0.topic_count`. However, trying to run this on a [testing website](https://jsonpath.curiousconcept.com/) generates a 403 error.

Does anyone know what's going wrong here? I'm not sure if there's a better way of testing these badges, since 'invalid' doesn't give much information.

Thanks

**Comment by chris48s** [comment, marked as answer]
Looks like forum.image.sc

a) bans certain user agents.
b) returns a plan text response instead of JSON if it doesn't like your user agent

```
$ curl "https://forum.image.sc/tag/cellprofiler.json" -A "Shields.io/2003a"
Crawler is not allowed!

$ curl "https://forum.image.sc/tag/cellprofiler.json"
Crawler is not allowed!

$ curl "https://forum.image.sc/tag/cellprofiler.json" -A "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:59.0) Gecko/20100101 Firefox/59.0"
{..snip - valid json..}
```

You'll have to take it up with the admins of forum.image.sc to see if you can get our UA unbanned in order to carry on using their API.

**Reply by DavidStirling** [reply]
Have had them add an exception for the shields domain, which fixed it. Thanks so much!
